src/shadowbox/server/manager_service.ts

```diff
@@ -205,54 +205,49 @@ export class ShadowsocksManagerService {
     try {
       logging.debug(`renameAccessKey request ${JSON.stringify(req.params)}`);
       const accessKeyId = req.params.id;
-      if (!this.accessKeys.renameAccessKey(accessKeyId, req.params.name)) {
-        return next(new restify.NotFoundError(`No access key found with id ${accessKeyId}`));
-      }
+      this.accessKeys.renameAccessKey(accessKeyId, req.params.name);
       res.send(HttpSuccess.NO_CONTENT);
       return next();
     } catch (error) {
       logging.error(error);
+      if (error instanceof errors.AccessKeyNotFound) {
+        return next(new restify.NotFoundError(error.message));
+      }
       return next(new restify.InternalServerError());
     }
   }
 
-  public async setAccessKeyQuota(req: RequestType, res: ResponseType, next: restify.Next) {
+  public async setAccessKeyLimit(req: RequestType, res: ResponseType, next: restify.Next) {
     try {
-      logging.debug(`setAccessKeyQuota request ${JSON.stringify(req.params)}`);
+      logging.debug(`setAccessKeyLimit request ${JSON.stringify(req.params)}`);
       const accessKeyId = req.params.id;
-      const quota = req.params.quota;
-      // TODO(alalama): remove these checks once the repository supports typed errors.
-      if (!quota || !quota.data || !quota.window) {
-        return next(new restify.InvalidArgumentError(
-            'Must provide a quota value with "data.bytes" and "window.hours"'));
-      }
-      if (quota.data.bytes < 0 || quota.window.hours < 0) {
-        return next(new restify.InvalidArgumentError('Must provide positive quota values'));
-      }
-      const success = await this.accessKeys.setAccessKeyQuota(accessKeyId, quota);
-      if (!success) {
-        return next(new restify.NotFoundError(`No access key found with id ${accessKeyId}`));
-      }
+      const limit = req.params.limit;
```

Comment:
Maybe use `restify.MissingParameterError` if the limit is missing? Still use status 400

Reply:
Done.